scripts/ci/gitlab/pipeline/build.yml

```diff
@@ -6,19 +6,28 @@
 .check-dependent-project:
   stage:                           build
   extends:
-    - .docker-env
+    - .kubernetes-env
```

Comment:
The job should run in docker-env on ci hosts in order to use rusty-cachier and more powerful hosts. Otherwise it takes 40 minutes to run which is too long

Reply:
After https://github.com/paritytech/pipeline-scripts/pull/70 the job won't do anything heavy because all the compilation and checking will move to the created pipeline instead of inside of the job directly.  Please check https://github.com/paritytech/opstooling/issues/128#issuecomment-1195171069 as well which also mentions the interaction with rusty-cachier.

The jobs run for a while, e.g. 40min of https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1701832, but it's because it was waiting for the created pipeline to be finished. The work of patching the branch and creating the pipeline takes less than 5 minutes.